brigade/core/inventory.py

```diff
@@ -192,12 +192,11 @@ def get_connection(self, connection):
         """
         if connection not in self.connections:
             task_name = "{}_connection".format(connection)
-            try:
-                task = getattr(connections, task_name)
-            except AttributeError:
-                raise AttributeError("not sure how to establish a connection for {}".format(
-                    connection))
-            task(host=self)
+            if kwargs.get("host") is None:
```

Comment:
I don't like this `if else` here. The core of brigade shouldn't care about specifics of tasks so whatever error handling a particular task needs should be handled in the task, not in the core.

Reply:
That is fine, will remove. 

I debated whether it made sense to raise an error here and thought it did as `host` is a pretty generic argument and both Netmiko and Brigade use `host` arguments.